pkg/apiserver/registry/stats/multicastgroup/rest.go

```diff
@@ -0,0 +1,142 @@
+	return &REST{p}
+}
+
+var (
+	_ rest.Storage = &REST{}
+	_ rest.Scoper  = &REST{}
+	_ rest.Getter  = &REST{}
+	_ rest.Lister  = &REST{}
+)
+
+type statsProvider interface {
+	ListMulticastGroups() []statsv1alpha1.MulticastGroup
+	GetMulticastGroup(name string) (*statsv1alpha1.MulticastGroup, bool)
+}
+
+func (r *REST) New() runtime.Object {
+	return &statsv1alpha1.MulticastGroup{}
+}
+
+func (r *REST) NewList() runtime.Object {
+	return &statsv1alpha1.MulticastGroupList{}
+}
+
+func (r *REST) List(ctx context.Context, options *internalversion.ListOptions) (runtime.Object, error) {
+	if !features.DefaultFeatureGate.Enabled(features.NetworkPolicyStats) {
+		return &statsv1alpha1.MulticastGroupList{}, nil
+	}
+	if !features.DefaultFeatureGate.Enabled(features.Multicast) {
+		return &statsv1alpha1.MulticastGroupList{}, nil
+	}
+	multicastGroups := &statsv1alpha1.MulticastGroupList{
+		Items: r.statsProvider.ListMulticastGroups(),
+	}
+	return multicastGroups, nil
+}
+
+func (r *REST) Get(ctx context.Context, name string, options *metav1.GetOptions) (runtime.Object, error) {
+	if !features.DefaultFeatureGate.Enabled(features.NetworkPolicyStats) {
+		return &statsv1alpha1.MulticastGroup{}, nil
+	}
+	if !features.DefaultFeatureGate.Enabled(features.Multicast) {
+		return &statsv1alpha1.MulticastGroup{}, nil
+	}
+	multicastGroup, exists := r.statsProvider.GetMulticastGroup(name)
+	if !exists {
+		return nil, errors.NewNotFound(statsv1alpha1.Resource("multicastgroup"), name)
+	}
+	return multicastGroup, nil
+}
+
+func (r *REST) ConvertToTable(ctx context.Context, obj runtime.Object, tableOptions runtime.Object) (*metav1.Table, error) {
+	table := &metav1.Table{
+		ColumnDefinitions: []metav1.TableColumnDefinition{
+			{Name: "Group", Type: "string", Format: "name", Description: "IP of multicast group."},
+			{Name: "Pods", Type: "string", Description: "List of Pods the has joined the multicast group."},
+		},
+	}
+	if m, err := meta.ListAccessor(obj); err == nil {
+		table.ResourceVersion = m.GetResourceVersion()
+		table.Continue = m.GetContinue()
+		table.RemainingItemCount = m.GetRemainingItemCount()
+	} else {
+		if m, err := meta.CommonAccessor(obj); err == nil {
+			table.ResourceVersion = m.GetResourceVersion()
+		}
+	}
+
+	var err error
+	table.Rows, err = metatable.MetaToTableRow(obj, func(obj runtime.Object, m metav1.Object, name, age string) ([]interface{}, error) {
+		stats := obj.(*statsv1alpha1.MulticastGroup)
+		return []interface{}{stats.Group, formatPodReferenceList(stats.Pods, 3)}, nil
+	})
+	return table, err
+}
+
+// formatPodReferenceList formats a list of PodReference and cut it if it encodes more than 3 pods.
+// Example: formatPodReferenceList(pods with len 13)
+// apodNamespace/apodName,bpodNamespace/bpodName,bpodNamespace/bpodName + 10 more...
+// The implementation of this func is similar to
+// https://github.com/kubernetes/kubernetes/blob/bb7b0b6a91e9f365025d496cb267817e777c872b/pkg/printers/internalversion/printers.go#L667.
```

Comment:
feel we don't need this pointer as the code is not very similar now and this is just a very normal util function

Reply:
updated